### PR TITLE
feat(mvp): implement handler for restore backup operation

### PR DESCRIPTION
**Pull request by ulya-sidorina**

### CODE REVIEW

File: internal/handlers/restore_backup.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+		)
+		return db.UpdateOperation(ctx, operation)
+	}
+
+	switch mr.State {
+	case types.OperationStatePending:
+		{
+			if !opResponse.GetOperation().Ready {
+				if (mr.CreatedAt.Unix() + config.OperationTtlSeconds) <= time.Now().Unix() {
+					response, err := client.CancelOperation(ctx, conn, mr.YdbOperationId)
```
Comment (qrort) [inline internal/handlers/restore_backup.go:88]: xlog.Info(cancelling operation due to ttl)

File: internal/types/backup.go [hunk 2]
```diff
@@ -83,6 +83,8 @@ type Operation interface {
 	SetState(s OperationState)
 	GetMessage() string
 	SetMessage(m string)
+	SetCreatedAt(createdAt time.Time)
```
Comment (qrort) [inline internal/types/backup.go:86]: i think we do not really need SetCreatedAt, GetCreatedAt in this interface. We only use them in restore handler => they can be fields of restore operation only

File: internal/types/backup.go [hunk 3]
```diff
@@ -120,13 +123,63 @@ func (o *TakeBackupOperation) GetMessage() string {
...
 	o.Message = m
 }
+func (o *TakeBackupOperation) GetCreatedAt() time.Time {
+	return o.CreatedAt
+}
+func (o *TakeBackupOperation) SetCreatedAt(createdAt time.Time) {
+	o.CreatedAt = createdAt
+}
+
+type RestoreBackupOperation struct {
```
Comment (qrort) [inline internal/types/backup.go:133]: It is also needed to correctly return `RestoreBackupOperation` instances from `process_result_set.go`

File: internal/handlers/restore_backup.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+		return fmt.Errorf(
+			"failed to get operation status for operation id %s, import operation id %s: %w",
+			mr.GetId().String(),
+			mr.YdbOperationId,
+			err,
+		)
+	}
+
+	if !isValidOperationResponse(opResponse) {
```
Comment (qrort) [inline internal/handlers/restore_backup.go:74]: Do we want to set Error if it is UNAVAILABLE or OVERLOADED? It is hard to say if we should Error or skip

File: cmd/ydbcp/main.go [hunk 5]
```diff
@@ -211,6 +211,15 @@ func main() {
 		return
 	}
 
+	err = handlersRegistry.Add(
+		types.OperationType("RB"), handlers.MakeRBOperationHandler(dbConnector, client.NewClientYdbConnector(), configInstance),
```
Comment (bma13) [inline cmd/ydbcp/main.go:215]: types.OperationTypeRB ?

File: cmd/ydbcp/main.go [hunk 6]
```diff
@@ -211,6 +211,15 @@ func main() {
 		return
 	}
 
+	err = handlersRegistry.Add(
+		types.OperationTypeRB, handlers.MakeRBOperationHandler(dbConnector, client.NewClientYdbConnector(), configInstance),
+	)
+
+	if err != nil {
+		log.Fatalf("failed to register RB handler: %v", err)
```
Comment (bma13) [inline cmd/ydbcp/main.go:219]: xlog ?

File: internal/connectors/client/connector.go [hunk 7]
```diff
@@ -179,3 +180,27 @@ func (d *ClientYdbConnector) ForgetOperation(ctx context.Context, clientDb *ydb.
...
+
+	response, err := client.CancelOperation(
+		ctx,
+		&Ydb_Operations.CancelOperationRequest{
+			Id: operationId,
+		},
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("error cancelling operation: %s", err.Error())
```
Comment (bma13) [inline internal/connectors/client/connector.go:202]: I'm not sure if we can see this error in the server log?
Let's log errors in grpc handlers

**Pull request merged**